website/src/Pages/ResourcesPage/AddResourcePage.jsx

```diff
@@ -0,0 +1,243 @@
+import { React, useState } from 'react';
+import {
+  Box, Button, Container, FormControl, MenuItem, Select, TextField, Typography,
+} from '@mui/material';
+import { useNavigate } from 'react-router-dom';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { TimePicker } from '@mui/x-date-pickers';
+import dayjs from 'dayjs';
+import { useSelector } from 'react-redux';
+import axios from 'axios';
+
+function AddResourcePage() {
+  const { authHeader } = useSelector((state) => state.sliceAuth);
+
+  const navigate = useNavigate();
+
+  const [title, setTitle] = useState('');
+  const [tag, setTag] = useState('');
+  const [days, setDays] = useState([0, 0, 0, 0, 0, 0, 0]);
+  const [phoneNumber, setPhoneNumber] = useState('');
+  const [startTime, setStartTime] = useState(dayjs());
+  const [endTime, setEndTime] = useState(dayjs().add(1, 'h'));
+  const [latitude, setLatitude] = useState('');
+  const [longitude, setLongitude] = useState('');
+  const [address, setAddress] = useState('');
+  const [email, setEmail] = useState('');
+  const [website, setWebsite] = useState('');
+
+  const updateDays = (num) => {
+    const updatedDaysStatus = [...days];
+    updatedDaysStatus[num] = updatedDaysStatus[num] ? 0 : 1;
+    setDays(updatedDaysStatus);
+  };
+
+  const backToResources = () => {
+    navigate('/resources');
+  };
+
+  const handlePhoneNumberChange = (event) => {
+    const inputValue = event.target.value;
+
+    // Remove all non-numeric characters from the input value
+    const numericValue = inputValue.replace(/\D/g, '');
+
+    // Format the phone number once it reaches 10 digits
+    if (numericValue.length === 10) {
+      const formattedValue = `(${numericValue.slice(0, 3)}) ${numericValue.slice(3, 6)}-${numericValue.slice(6, 10)}`;
+      setPhoneNumber(formattedValue);
+    } else if (numericValue.length < 10) {
+      setPhoneNumber(numericValue);
+    }
+  };
+
+  const createResource = async () => {
```

Comment:
add validation for fields, if something fails maybe we could display a message of what failed

Reply:
again, I will leave the alert and validation to the frontend but I implemented preliminary checks @zhangjuliet